src/OpenBoleto/BoletoAbstract.php

```diff
@@ -1,1697 +1,1725 @@
+     * @var string
+     */
+    protected $especieDoc;
+
+    /**
+     * Número do documento
+     * @var int
+     */
+    protected $numeroDocumento;
+
+    /**
+     * Número da parcelas
+     * @var string
+     */
+    protected $numParcelas;
+
+    /**
+     * Define o número sequencial definido pelo cliente para compor o Nosso Número
+     *
+     * @var int
+     */
+    protected $sequencial;
+
+    /**
+     * Campo de uso do banco no boleto
+     * @var string
+     */
+    protected $usoBanco;
+
+    /**
+     * Agência
+     * @var int
+     */
+    protected $agencia;
+
+    /**
+     * Dígito da agência
+     * @var string|int
+     */
+    protected $agenciaDv;
+
+    /**
+     * Conta
+     * @var int|string
+     */
+    protected $conta;
+
+    /**
+     * Dígito da conta
+     * @var int
+     */
+    protected $contaDv;
+
+    /**
+     * Modalidade de cobrança do cliente, geralmente Cobrança Simples ou Registrada
+     * @var string
+     */
+    protected $carteira;
+
+    /**
+     * Define as carteiras disponíveis para cada banco
+     * @var array
+     */
+    protected $carteiras = array();
+
+    /**
+     * Define as carteiras disponíveis para cada banco
+     * @var array
+     */
+    protected $carteirasNomes = array();
+
+    /**
+     * Entidade cedente (quem emite o boleto)
+     * @var \OpenBoleto\Agente
+     */
+    protected $cedente;
+
+    /**
+     * Entidade sacada (de quem se cobra o boleto)
+     * @var \OpenBoleto\Agente
+     */
+    protected $sacado;
+
+    /**
+     * Entidade sacador avalista
+     * @var \OpenBoleto\Agente
+     */
+    protected $sacadorAvalista;
+
+    /**
+     * Array com as linhas do demonstrativo (descrição do pagamento)
+     * @var array
+     */
+    protected $descricaoDemonstrativo;
+
+    /**
+     * Linha de local de pagamento
+     * @var string
+     */
+    protected $localPagamento = 'Pagável em qualquer agência bancária até o vencimento.';
+
+    /**
+     * Array com as linhas de instruções
+     * @var array
+     */
+    protected $instrucoes = array('Pagar até a data do vencimento.');
+
+    /**
+     * Nome do arquivo de template a ser usado
+     * @var string
+     */
+    protected $layout = 'default.phtml';
+
+    /**
+     * Pasta de localização de resources (imagens, css e views)
+     * @var string
+     */
+    protected $resourcePath;
+
+    /**
+     * Localização do logotipo da empresa
+     * @var string
+     */
+    protected $logoPath;
+
+    /**
+     * Localização do logotipo do banco, referente ao diretório de imagens
+     * @var string
+     */
+    protected $logoBanco;
+
+    /**
+     * Array que sera exportada pelo metodo getData
+     * @var array
+     */
+    protected $data;
+
+    /**
+     * Imprime ou não as instruções de impressão
+     * @var bool
+     */
+    protected $imprimeInstrucoesImpressao = true;
+
+    /**
+     * QrCode do boleto
+     *
+     * @return string
```

Comment:
Aqui seria `@var string` e aí inicializar` $qrCode` com string vazia ou usar `@var string|null`

Reply:
Ou remover esse @varjá que foi usado tipagem nativa.
Na minha próxima contribuição, vamos remover annotations e usar tipagem nativa onde for possível
No caso ficaria `protected string $qrCote = '';`

Reply:
ou `protected ?string $qrCote = null;`